src/rgw/rgw_oidc_provider.cc

```diff
@@ -1,182 +1,41 @@
 #include "rgw_oidc_provider.h"
 
-#include "services/svc_zone.h"
-#include "services/svc_sys_obj.h"
-
 #define dout_subsys ceph_subsys_rgw
 
-using namespace std;
-
-namespace rgw { namespace sal {
-
-const string RGWOIDCProvider::oidc_url_oid_prefix = "oidc_url.";
-const string RGWOIDCProvider::oidc_arn_prefix = "arn:aws:iam::";
-
-int RGWOIDCProvider::get_tenant_url_from_arn(string& tenant, string& url)
-{
-  auto provider_arn = rgw::ARN::parse(arn);
-  if (!provider_arn) {
-    return -EINVAL;
-  }
-  url = provider_arn->resource;
-  tenant = provider_arn->account;
-  auto pos = url.find("oidc-provider/");
-  if (pos != std::string::npos) {
-    url.erase(pos, 14);
-  }
-  return 0;
-}
-
-int RGWOIDCProvider::create(const DoutPrefixProvider *dpp, bool exclusive, optional_yield y)
-{
-  int ret;
-
-  if (! validate_input(dpp)) {
-    return -EINVAL;
-  }
-
-  string idp_url = url_remove_prefix(provider_url);
-
-  /* check to see the name is not used */
-  ret = read_url(dpp, idp_url, tenant, y);
-  if (exclusive && ret == 0) {
-    ldpp_dout(dpp, 0) << "ERROR: url " << provider_url << " already in use"
-                    << id << dendl;
-    return -EEXIST;
-  } else if ( ret < 0 && ret != -ENOENT) {
-    ldpp_dout(dpp, 0) << "failed reading provider url  " << provider_url << ": "
-                  << cpp_strerror(-ret) << dendl;
-    return ret;
-  }
-
-  //arn
-  arn = oidc_arn_prefix + tenant + ":oidc-provider/" + idp_url;
-
-  // Creation time
-  real_clock::time_point t = real_clock::now();
-
-  struct timeval tv;
-  real_clock::to_timeval(t, tv);
-
-  char buf[30];
-  struct tm result;
-  gmtime_r(&tv.tv_sec, &result);
-  strftime(buf,30,"%Y-%m-%dT%H:%M:%S", &result);
-  sprintf(buf + strlen(buf),".%dZ",(int)tv.tv_usec/1000);
-  creation_date.assign(buf, strlen(buf));
-
-  ret = store_url(dpp, idp_url, exclusive, y);
-  if (ret < 0) {
-    ldpp_dout(dpp, 0) << "ERROR:  storing role info in OIDC pool: "
-                  << provider_url << ": " << cpp_strerror(-ret) << dendl;
-    return ret;
-  }
-
-  return 0;
-}
-
-int RGWOIDCProvider::get(const DoutPrefixProvider *dpp, optional_yield y)
-{
-  string url, tenant;
-  auto ret = get_tenant_url_from_arn(tenant, url);
-  if (ret < 0) {
-    ldpp_dout(dpp, 0) << "ERROR: failed to parse arn" << dendl;
-    return -EINVAL;
-  }
-
-  if (this->tenant != tenant) {
-    ldpp_dout(dpp, 0) << "ERROR: tenant in arn doesn't match that of user " << this->tenant << ", "
-                  << tenant << ": " << dendl;
-    return -EINVAL;
-  }
-
-  ret = read_url(dpp, url, tenant, y);
-  if (ret < 0) {
-    return ret;
-  }
-
-  return 0;
-}
-
-void RGWOIDCProvider::dump(Formatter *f) const
+void RGWOIDCProviderInfo::dump(Formatter *f) const
 {
-  encode_json("OpenIDConnectProviderArn", arn, f);
+  encode_json("id", id, f);
+  encode_json("provider_url", provider_url, f);
+  encode_json("arn", arn, f);
+  encode_json("creation_date", creation_date, f);
+  encode_json("tenant", tenant, f);
```

Comment:
do we need a check here that if the oidc provider belongs to a tenant, then display it, else don't

Reply:
ok, got it - `tenant` represents both account and tenant